baked_in.go

```diff
@@ -527,7 +530,28 @@ func isEthereumAddress(fl FieldLevel) bool {
 		return true
 	}
 
-	// checksum validation is blocked by https://github.com/golang/crypto/pull/28
+	// Checksum validation. Reference: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+	digitValue := map[byte]int{
+		'0': 0, '1': 1, '2': 2, '3': 3, '4': 4,
+		'5': 5, '6': 6, '7': 7, '8': 8, '9': 9,
+		'a': 10, 'b': 11, 'c': 12,
+		'd': 13, 'e': 14, 'f': 15,
+	}
+	address = address[2:]
+	h := sha3.NewLegacyKeccak256()
+	h.Write([]byte(strings.ToLower(address)))
+	hash := hex.EncodeToString(h.Sum(nil))
+
+	for i := 0; i < len(address); i++ {
```

Comment:
Here `len(address)` definitely has value 40, but I preferred to write `len(address)` instead of `40`. Thus we keep the information that "Ethereum addresses have 40 hex digits" declared only in one place, in the `regexes.go` (`ethAddress{Upper,Lower,}RegexString`).

Let me know if otherwise